daemon_unix.go

```diff
@@ -115,6 +116,71 @@ func (d *Context) parent() (child *os.Process, err error) {
 	d.rpipe.Close()
 	encoder := json.NewEncoder(d.wpipe)
 	err = encoder.Encode(d)
+
+	// wait for worker to start running or else network calls fail: https://github.com/sevlyar/go-daemon/issues/100
+	time.Sleep(2 * time.Second)
+
+	return
+}
+
+var initialized = false
+
+func (d *Context) child() (err error) {
+	if initialized {
+		return os.ErrInvalid
+	}
+	initialized = true
+
+	decoder := json.NewDecoder(os.Stdin)
+	if err = decoder.Decode(d); err != nil {
+		return
+	}
+
+	// create PID file after context decoding to know PID file full path.
+	if len(d.PidFileName) > 0 {
+		d.pidFile = NewLockFile(os.NewFile(4, d.PidFileName))
+		if err = d.pidFile.WritePid(); err != nil {
+			return
+		}
+		defer func() {
+			if err != nil {
+				d.pidFile.Remove()
+			}
+		}()
+	}
+
+	if err = syscallDup(3, 0); err != nil {
+		return
+	}
+
+	if d.Umask != 0 {
+		syscall.Umask(int(d.Umask))
```

Comment:
Looks like the PR contains a lot of changes. And i should review all of them. But really the PR contains the one-line fix and other it is just moves of the old code. Please return it back or provide the refactoring (code moves) in a separate PR with appropriate comment.